paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/kafka/KafkaSchema.java

```diff
@@ -99,7 +100,15 @@ private static KafkaConsumer<String, String> getKafkaEarliestConsumer(
         }
         int firstPartition =
                 partitionInfos.stream().map(PartitionInfo::partition).sorted().findFirst().get();
-        consumer.assign(Collections.singletonList(new TopicPartition(topic, firstPartition)));
+        Collection<TopicPartition> topicPartitions =
+                Collections.singletonList(new TopicPartition(topic, firstPartition));
+        consumer.assign(topicPartitions);
+
+        Map<TopicPartition, Long> beginningOffsets = consumer.beginningOffsets(topicPartitions);
+        for (TopicPartition tp : topicPartitions) {
+            Long offset = beginningOffsets.get(tp);
+            consumer.seek(tp, offset);
+        }
```

Comment:
How about just using `consumer.seekToBeginning(topicPartitions);` ?

Reply:
> How about just using `consumer.seekToBeginning(topicPartitions);` ?

Yes, that's more straight forward, adopt your suggestion.